File: RegexFileSorter/Types/ObservableObject.vb
Imports System.ComponentModel
Imports System.Runtime.CompilerServices

Public Class ObservableObject
	Implements INotifyPropertyChanged

	Protected Sub OnPropertyChanged(<CallerMemberName()> Optional ByVal propertyName As String = Nothing)
		RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
	End Sub

	Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

End Class

File: RegexFileSorter/ObservableObject.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace RegexFileSorter
{
    internal class ObservableObject : INotifyPropertyChanged
    {
        protected void OnPropertyChanged([CallerMemberName] string propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        public event PropertyChangedEventHandler PropertyChanged;
    }
}